fighters/packun/src/opff.rs

```diff
@@ -4,22 +4,153 @@ use super::*;
+            apply_status_speed_mul(fighter, 0.75);
+        }
+        VarModule::off_flag(fighter.object(), vars::packun::instance::STANCE_NEED_SET_SPEEDS);
+    }
+
+    if fighter.status() != VarModule::get_int(fighter.object(), vars::packun::instance::STANCE_STATUS) {
+        //println!("Status is changing!");
+        VarModule::on_flag(fighter.object(), vars::packun::instance::STANCE_NEED_SET_SPEEDS);
+        VarModule::set_int(fighter.object(), vars::packun::instance::STANCE_STATUS, fighter.status());
+        //println!("new stance status: {}", VarModule::get_int(fighter.object(), vars::packun::instance::STANCE_STATUS));
+    }
+
+    // dash & momentum transfer speeds
+    if VarModule::get_int(fighter.object(), vars::packun::instance::CURRENT_STANCE) == 1 {
+        VarModule::set_float(fighter.object(), vars::common::instance::JUMP_SPEED_MAX_MUL, 0.88);
+
+        // if you are initial dash, slow them down slightly
+        if fighter.is_status(*FIGHTER_STATUS_KIND_DASH) {
+            let motion_vec = Vector3f {
+                x: -0.15 * PostureModule::lr(fighter.boma()) * (1.0 - (MotionModule::frame(fighter.boma()) / MotionModule::end_frame(fighter.boma()))),
+                y: 0.0, 
+                z: 0.0
+            };
+            KineticModule::add_speed_outside(fighter.boma(), *KINETIC_OUTSIDE_ENERGY_TYPE_WIND_NO_ADDITION, &motion_vec);
+        }
+    }
+
+    else if VarModule::get_int(fighter.object(), vars::packun::instance::CURRENT_STANCE) == 2 {
+        VarModule::set_float(fighter.object(), vars::common::instance::JUMP_SPEED_MAX_MUL, 0.75);
+
+        // if you are initial dash, slow them down slightly
+        if fighter.is_status(*FIGHTER_STATUS_KIND_DASH) {
+            let motion_vec = Vector3f {
+                x: -0.25 * PostureModule::lr(fighter.boma()) * (1.0 - (MotionModule::frame(fighter.boma()) / MotionModule::end_frame(fighter.boma()))),
+                y: 0.0, 
+                z: 0.0
+            };
+            KineticModule::add_speed_outside(fighter.boma(), *KINETIC_OUTSIDE_ENERGY_TYPE_WIND_NO_ADDITION, &motion_vec);
+        }
+    }
+}
+
+/// checks if stance should be reset due to death or match end
+unsafe fn check_reset(fighter: &mut smash::lua2cpp::L2CFighterCommon) {
+    if fighter.is_status_one_of(&[
+        *FIGHTER_STATUS_KIND_WIN,
+        *FIGHTER_STATUS_KIND_LOSE,
+        *FIGHTER_STATUS_KIND_ENTRY,
+        *FIGHTER_STATUS_KIND_DEAD]) {
+            VarModule::set_int(fighter.object(), vars::packun::instance::CURRENT_STANCE, 0);
+    }
+}
+
+/// applies the given multiplier to various speed stats of the given fighter. 
+/// This should only be called once per status, or you will get some multiplicative effects
+unsafe fn apply_status_speed_mul(fighter: &mut smash::lua2cpp::L2CFighterCommon, mul: f32) {
+    // set the X motion speed multiplier (where movement is baked into an anim)
+    lua_bind::FighterKineticEnergyMotion::set_speed_mul(fighter.get_motion_energy(), mul);
```

Comment:
Please do
`
fighter.clear_lua_stack();
lua_args!(fighter, FIGHTER_KINETIC_ENERGY_ID_MOTION);
let og_speed_mul = app::sv_kinetic_energy::get_speed_mul(fighter.lua_state_agent);
lua_bind::FighterKineticEnergyMotion::set_speed_mul(fighter.get_motion_energy(), og_speed_mul * mul);
`
instead. This allows dash attack to be naturally affected by your status speed mul.